crates/lsp/src/lsp.rs

```diff
@@ -92,13 +92,21 @@ pub enum Subscription {
     },
 }
 
+/// ID is either string or number
```

Comment:
This rustdoc does not make much sense, as repeats exactly what this `enum` declares.
I would propose to link the spec here instead, just like `see <a href="https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#requestMessage">the specification</a> for possible values`.

Thinking of this more, the integer id should be `i32` actually, not `usize`? 
The spec defines https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#uinteger for unsized ints and this is not our case.

If we want to make it right, we should change to that type instead?

Reply:
Updated the rust doc, and as a second commit changed the ID type to i32.